torch/backends/cudnn/__init__.py

```diff
@@ -12,34 +12,37 @@
 #
 #   torch.backends.cudnn.enabled = False
 #
-# to globally disable CuDNN
+# to globally disable CuDNN/MIOpen
 
 lib = None
 __cudnn_version = None
 # TODO: dynamic version checks via cudnnGetVersion
 
 def find_cudnn_windows_lib():
-    # Override the default search process
-    # Fixes https://github.com/pytorch/pytorch/issues/20202
-    # The library selection will be done in these directories one by one
-    # 1. [Package Root]\Lib 
-    #    That's where our libraries are in, which should be loaded first.
-    # 2. Default directories
-    #    That is stored in the environment variable `PATH`.
-    test_env = os.environ.copy()
-    old_path = test_env['PATH']
-    th_dll_path = os.path.join(os.path.dirname(
-        os.path.dirname(os.path.dirname(__file__))), 'lib')
-    test_env['PATH'] = ';'.join([th_dll_path, old_path])
-    proc = Popen(['where', 'cudnn64*.dll'], stdout=PIPE,
-                 stderr=PIPE, stdin=PIPE, env=test_env)
-    out, err = proc.communicate()
-    out = out.decode().strip()
-    if len(out) > 0:
-        if out.find('\r\n') != -1:
-            out = out.split('\r\n')[0]
-        cudnn_lib = str(out)
-        return ctypes.cdll.LoadLibrary(cudnn_lib)
+    if cuda is not None:
```

Comment:
Currently we don't have hip build on windows

Reply:
Right, that's exactly why I put the cuda guard, so it returns None for hip. The alternative was to not define/call the function, but I thought this would be more future-proof if, at a later point in time, we had a hip build on windows. How would you prefer it to be handled?

Reply:
I would suggest let's not make the changes in this diff and re-think about windows build later.

Reply:
That's fair. I see now what you meant by the absence of a hip build for windows: the `find_cudnn_windows_lib` function wouldn't even be called for hip.